sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedParquetRecordReader.java

```diff
@@ -90,6 +91,8 @@
    */
   private ColumnarBatch columnarBatch;
 
+  private MemoryMode memoryMode = null;
```

Comment:
This is because `initBatch()` has to distinguish two cases. 
1. `initBatch()` is called without calling `initBatch(partitionColumns, partitionValues, isEnabledOffHeap)`
2. `initBatch()` is called after `initBatch(partitionColumns, partitionValues, isEnabledOffHeap)` was called.

In the case 2., we have to keep the memory mode that has been set by `initBatch(partitionColumns, partitionValues, isEnabledOffHeap)`.